pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -167,6 +174,66 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
+	nodeVolumeMaps map[string]map[types.NodeName][]*volume.Spec,
+	volumeSpecMap map[*volume.Spec]v1.UniqueVolumeName,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+
+	return func() error {
+		for pluginName, pluginNodeVolumes := range nodeVolumeMaps {
+			attachableVolumePlugin, err :=
+				og.volumePluginMgr.FindAttachablePluginByName(pluginName)
+			if err != nil || attachableVolumePlugin == nil {
+				glog.Errorf(
+					"BulkVerifyVolume.FindAttachablePluginBySpec failed for plugin %q with: %v",
+					pluginName,
+					err)
+				continue
+			}
+
+			volumeAttacher, newAttacherErr := attachableVolumePlugin.NewAttacher()
+
+			if newAttacherErr != nil {
+				glog.Errorf(
+					"BulkVerifyVolumes failed for getting plugin %q with: %v",
+					attachableVolumePlugin,
+					newAttacherErr)
+				continue
+			}
+			bulkVolumeVerifier, ok := volumeAttacher.(volume.BulkVolumeVerifier)
+
+			if !ok {
+				glog.Errorf("BulkVerifyVolume failed to type assert attacher %q", bulkVolumeVerifier)
+				continue
+			}
+
+			attached, bulkAttachErr := bulkVolumeVerifier.BulkVerifyVolumes(pluginNodeVolumes)
+			if bulkAttachErr != nil {
+				glog.Errorf("Error checking volumes are attached with %v", bulkAttachErr)
+				continue
+			}
+
+			for nodeName, volumeSpecs := range pluginNodeVolumes {
+				for _, volumeSpec := range volumeSpecs {
+					nodeVolumeSpecs, nodeChecked := attached[nodeName]
+
+					if !nodeChecked {
+						actualStateOfWorld.MarkVolumeAsDetached(volumeSpecMap[volumeSpec], nodeName)
```

Comment:
missing `continue`?

I think you could also start with `check := false`, and then update it `if nodeVolumeSpecs != nil`